server/transaction/chunk.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transaction
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/tigrisdata/tigris/internal"
+	"github.com/tigrisdata/tigris/keys"
+	"github.com/tigrisdata/tigris/store/kv"
+)
+
+const (
+	KB   = 1000
+	KB99 = 99 * KB
+
+	chunkIdentifier = "__"
+)
+
+var chunkSize = KB99
+
+type chunkCB func(chunkNo int32, chunkData []byte) error
+
+// ChunkTx will be used to split the user payload if it is greater than 99KB. It adds some metadata in the zeroth chunk
+// so that it can rebuild the body again. The chunk number is only appended in the first chunk as we don't need to do
+// anything for zeroth chunk. The chunk number is appended at the end in the format "__<chunk number>".
```

Comment:
I'm wondering if we should call it first chunk over `zeroth` chunck

Reply:
:) yeah, either way works. But mainly wanted to avoid confusion that the chunk number will not be `1` for this chunk, either it would be `nil` or `0`